libraries/core/src/main/kotlin/com/bumble/appyx/core/store/RetainedInstanceStoreImpl.kt

```diff
@@ -11,6 +11,22 @@ internal class RetainedInstanceStoreImpl : RetainedInstanceStore {
             .getOrPut(key) { ValueHolder(factory(), disposer) }
             .value as T
 
+    override fun isRetainedByStoreId(storeId: String, value: Any): Boolean =
+        map[storeId]
+            ?.let { valueHolderMap ->
```

Comment:
[nit] could be a reusable function for both implementations
```
fun MutableMap<String, ValueHolder<*>.hasValue(value: Any) {
    values.any { valueHolder ->
         valueHolder.value === value
    }
}
```

Reply:
Good suggestion. I have applied it with a couple of tweaks